### PR TITLE
Fix build error caused by content in datasource guide

### PR DESCRIPTION
**Pull request by rolfedh**
In community `main`, line 550 in `/asciidoc/datasource.adoc` is:

```
<<quarkus-agroal_quarkus-datasource-jdbc-transactions,`quarkus.datasource[.optional name].jdbc.transactions`>> to `xa
```
In the syncrhonization PR, it causes the following preview build failure, 
```
Transforming the AsciiDoc content to DocBook XML... [31m[1mOpening and ending tag mismatch: literal line 566 and link, line 566, column 66[0m [31m[1mUnable to parse the AsciiDoc built DocBook XML[0m
```
In this PR, the previously mentioned line 550 has been transformed into:
```
xref:quarkus-agroal_quarkus-datasource-jdbc-transactions[`quarkus.datasource[.optional name].jdbc.transactions`] to `xa`.
```
If you inspect line 556 in the master.adoc file for this PR, https://jenkins.dxp.redhat.com/job/CCS/job/ccs-mr-preview/80313/artifact/pantheon/configure-data-sources/build/en-US/master.xml/*view*/, that same line renders as follows:
```
 <link linkend="quarkus-agroal_quarkus-datasource-jdbc-transactions"><literal>quarkus.datasource[.optional name</link>.jdbc.transactions</literal>] to <literal>xa</literal>.</simpara> </listitem>
```
In other words, the right bracket in `name]` terminates the `<link>` element prematurely. Instead, the right bracket in `</literal>]` should close that `<link>` element. 

I have confirmed that escaping the right bracket in `name]` solves the issue, as shown in the following example:

```
xref:quarkus-agroal_quarkus-datasource-jdbc-transactions[`quarkus.datasource[.optional name\].jdbc.transactions`] 
```

### CODE REVIEW

**Comment by rolfedh** [comment]
Cross reference: https://issues.redhat.com/browse/QDOCS-789

**Comment by MichalMaler** [comment]
@rolfedh Ok. lets try that out. Thank you for the PR. :)

**Review by MichalMaler — APPROVED** [review]

**Comment by MichalMaler** [comment]
I guess this is not something generated, thus the escape should not ruin anything, but I am not 100% sure.
I asked @jmartisk and he thinks that the best option would be to have an option to run our downstreaming pipeline on this PR directly; before merging.

**Comment by jmartisk** [comment]
I looked at the (upstream) documentation built from this PR and the backslash visibly appears visibly in the HTML. Screenshot:

image

So this doesn't look entirely correct to me, is there another solution?

**Comment by jmartisk** [comment]
Would replacing the `[]` with either `""` or `<>`be reasonable? I think so, and perhaps it would fix the downstream build

**Comment by gsmet** [comment]
I think it's because we should only escape the character when preparing the downstream doc.

**Comment by gsmet** [comment]
@rolfedh do you confirm that escaping the `]` fixes the issue with the downstream tooling and that you don't see it in the output? If so I will adjust the downstream script.

**Comment by MichalMaler** [comment]
@gsmet Hello Guillame. If @rolfedh will not be 100% sure about the fix, can we merge this, test the pipeline, and then revert it?
My apologies, but I am not sure right now hot to test it properly. Also, we dont want to add more load on you if we are not 100% sure.

**Comment by MichalMaler** [comment]
@jmartisk Thank you, Jan. You are just great. I am on PTO since next few hours, but I will let @rolfedh and we can progress if this is all good in @gsmet books.

**Comment by rolfedh** [comment]
Hi, @gsmet. Thanks for asking. Yes, I verified that that escaping the square bracket in ` name]` resolves the issue by performing a downstream build using bccutils, the same toolchain that PV1 uses. I inspected the xml and html outputs and see that the link element is correctly formed with that one change. Please go ahead then and do what you propose.

**Comment by rolfedh** [comment]
> Turns out the docs team currently doesn't have an easy way to test it without merging it.

Good point, @jmartisk.

I did some research and found a potential workaround for this problem. If a writer encounters a build failure in a synchronization job PR, they can use the [GitHub `gh` CLI](https://cli.github.com/manual/gh_pr_checkout) to check out the PR as a local branch to test potential fixes. Once they verify the fix locally, they can submit that fix to the community main branch and repeat the synchronization pipeline job.

**Comment by gsmet** [comment]
I have some local tests running atm so can't work on that but I'll have a look tomorrow morning.

**Comment by rolfedh** [comment]
> I have some local tests running atm so can't work on that but I'll have a look tomorrow morning.

Thank you, @gsmet!

**Comment by jmartisk** [comment]
> > Turns out the docs team currently doesn't have an easy way to test it without merging it.
> 
> Good point, @jmartisk.
> 
> I did some research and found a solution so writers don't have to perform "dirty" merges. If a writer encounters a build failure in a synchronization job PR, they can use [these git commands](https://stackoverflow.com/questions/27567846/how-can-i-check-out-a-github-pull-request-with-git) or the GitHub CLI [`gh pr checkout`](https://cli.github.com/manual/gh_pr_checkout) command to check out the PR as a local branch. There, they can perform local builds to inspect the failure and test potential fixes. Once they verify the fix locally, they can submit that fix to the community main branch and repeat the synchronization pipeline job.

If that works for you then great. What I had in mind was the ability to make "dry runs" of the synchronization pipeline that would be based on a branch of a particular (upstream) PR, instead of a regular branch like `main`. Then you could see immediately that an upstream PR changes something that would break the synchronization job later. If you later decide that this would be useful to you, I am able to help (it would require parametrizing some additional stuff in the downstream scripts).

**Comment by MichalMaler** [comment]
@jmartisk That would be great. Let me have a chat with Rolfe on Monday to decide about this and we will get back to you.
I will be happy to help in any way.
Thanks and kind regards. Cheers!

**Comment by rolfedh** [comment]
@MichalMaler @jmartisk I invited Mickey to discuss this in about an hour or so.
